components/config/opts.rs

```diff
@@ -667,13 +666,12 @@ pub fn from_cmdline_args(args: &[String]) -> ArgumentParsingResult {
 
     let url = match url_opt {
         Some(url_string) => {
-            parse_url_or_filename(&cwd, url_string)
-                .unwrap_or_else(|()| args_fail("URL parsing failed"))
+            match parse_url_or_filename(&cwd, url_string) {
+                Ok(v) => Some(v),
+                Err(_) => None,
+            }
         },
-        None => {
-            print_usage(app_name, &opts);
-            args_fail("servo asks that you provide a URL")
-        }
+        None => None,
     };
```

Comment:
I think you can combine all of this with `and_then`, something like this:

```rust
let url_opt = url_opt.and_then(|url_string| parse_url_or_filename(…).ok());
```

Also, rename `url` to `url_opt` (still an option at this point)